core/include/scipp/core/dimensions.h

```diff
@@ -146,9 +146,20 @@ Dimensions merge(const Dimensions &a, const Dimensions &b,
   return merge(merge(a, b), other...);
 }
 
+SCIPP_CORE_EXPORT Dimensions intersection(const Dimensions &a,
+                                          const Dimensions &b);
+
 SCIPP_CORE_EXPORT Dimensions transpose(const Dimensions &dims,
                                        std::vector<Dim> labels = {});
 
+SCIPP_CORE_EXPORT Dimensions fold(const Dimensions &old_dims,
+                                  const Dim from_dim,
+                                  const Dimensions &to_dims);
+
+SCIPP_CORE_EXPORT Dimensions flatten(const Dimensions &old_dims,
+                                     const scipp::span<const Dim> from_labels,
+                                     const Dim to_dim);
```

Comment:
Should use `[[nodiscard]]`.

Reply:
Does this mean I have to add `[[nodiscard]]` to `transpose` and `merge` and all the other functions in the `shape.h` files?